test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,154 +130,100 @@ public ServerSocket createServerSocket(int port) throws IOException {
-
         switch (test) {
-        case 1: /* default constructor - default config */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory());
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
-            }
-            break;
-        case 2: /* non-default constructor - default config */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            null,
-                                            false));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
-            }
-            break;
-        case 3: /* needClientAuth=true */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            null,
-                                            null,
-                                            true));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
+            /* default constructor - default config */
+            case 1 -> run(new ServerFactory(), false);
+
+            /* non-default constructor - default config */
+            case 2 -> run(new ServerFactory(null, null, false), false);
+
+            /* needClientAuth=true */
+            case 3 -> run(new ServerFactory(null, null, null, true), false);
+
+            /* server side dummy_ciphersuite */
+            case 4 -> {
+                try {
+                    new ServerFactory(SSLContext.getDefault(),
+                            new String[]{"dummy_ciphersuite"}, null, false);
```

Comment:
what if the constructor doesn't throw any exception? shouldn't the test fail? or should `run` be called in that case?

Reply:
good catch; yes, the test should fail. I updated these cases.